templates/todo/projects/nodejs-mongo-aks/.repo/bicep/azure.yaml

```diff
@@ -17,6 +17,21 @@ services:
     dist: dist
     language: js
     host: aks
+    hooks:
+      prebuild:
```

Comment:
nit.  lifecycle hooks are cool, but they are very couple to the service-framework.  Each hook could mean something different, for example, if you are using a composed framework like docker or swa, or if you are using a compiling language like C# or a non-compile language like JS (the case for this template).

Since `todo-templates` are references, I would try to avoid creating this coupling between the specifics of the template and the lifecycle hooks.  Instead, I would use the command hook `predeploy` and `postdeploy`, which would be consistent with all other `todo` teamplates examples.  It brings less things to understand for customers while still adding the functionality we want.

Reply:
In my opinion any of the VITE template updates should be using the `web` services `prebuild` hook to ensure the environment variables are created in the correct order.  All of todo templates are based on the same web front end Node based project.

`API_BASE_URL` is required to be set before the container image is built for the `web` service to ensure the correct URL is built into the container image. `predeploy` happens too late because at that point in time the container has already been built and you incorrectly would be setting an empty value.